### PR TITLE
Added more documentation and fixed updateMirror bug

### PR DESCRIPTION
**Pull request by soapsrc**

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -103,31 +112,36 @@ function scaleToFit(img) {
...
+    // Draw initial platform
     ctx.drawImage(pf, pfX, pfY, pf.width, pf.height);
-    // amount to move
+    // Distance x-coordinate of platform is moved
     pfX += dx;
 }
-
+/**
+ * Returns None
+ * Draws chef Kirby, walking Kirby
```
Comment (mightymosdan) [inline js/script.js]: Doesn't this draw the mirror with the currentmeme?

File: js/script.js [hunk 2]
```diff
@@ -179,19 +196,23 @@ function loadArray(kirby) {
...
     itemX = (canvas.width / 2.5) + ckirby.width; // X coordinate of item
     itemY = 200 + (ckirby.height); // Y coordinate of item
 }
-
-// Checks if Kirby has collided with current item (only one item can be called at a time or function will not work)
+/**
+ * Returns None
+ * Checks if Kirby has collided with current item (only one item can be called at a time or function will not work)
+ */
 function notCollided() {
```
Comment (mightymosdan) [inline js/script.js]: Can this be renamed to checkCollision ?

File: js/script.js [hunk 3]
```diff
@@ -202,58 +223,82 @@ function notCollided() {
...
+        newrandommeme = memearray[arrayIndex];
     }
+    // Update  currentmeme
     currentmeme = newrandommeme;
 
 }
-
+/**
+ * Returns None
+ * Handles double click on chef Kirby by triggering tomato animation and moving it from pot -> platform
```
Comment (mightymosdan) [inline js/script.js:298]: Should this say that this function handles clicks and item animations?

File: js/script.js (hunk 2 above)
Comment (soapsrc) [reply js/script.js]: Yes good point

File: js/script.js (hunk 1 above)
Comment (soapsrc) [reply js/script.js]: Oops I will edit this comment.

File: js/script.js (hunk 3 above)
Comment (soapsrc) [reply js/script.js:298]: Yes I just wrote a more elaborated description to make it more clear.

**Pull request merged**